tests/functional_tests/0003_create_test.py

```diff
@@ -21,30 +21,225 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 # IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
-import re
+
+import os
+import json
+import tempfile
+import uuid
 
 import pytest
-from click.testing import CliRunner
 
-import iocage_cli as ioc
 
 require_root = pytest.mark.require_root
 require_zpool = pytest.mark.require_zpool
+require_dhcp = pytest.mark.require_dhcp
+require_networking = pytest.mark.require_networking
+
+# LIST OF FLAGS in create TO TEST
+# [c, C , r, t, p, n, u, b, T, e, s, f]
+
+# TODO: Test different jail props as well
+
+
+@require_root
+@require_zpool
+def test_01_create_jail_with_uuid(release, invoke_cli, jail):
+    u_flag = str(uuid.uuid4())
+
+    invoke_cli(
+        ['create', '-r', release, '-u', u_flag]
+    )
+
+    assert jail(u_flag).exists is True
+
+
+@require_root
+@require_zpool
+def test_02_create_jail_with_name(release, jail, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'name_jail']
+    )
+
+    assert jail('name_jail').exists is True
+
+
+@require_root
+@require_zpool
+def test_03_create_jail_with_uuid_and_short(release, jail, invoke_cli):
+    u_flag = str(uuid.uuid4())
+
+    invoke_cli(
+        ['create', '-r', release, '-s', '-u', u_flag]
+    )
+
+    assert jail(u_flag[:8]).exists is True
+
+
+@require_root
+@require_zpool
+def test_04_create_count_jails_with_name(release, jail, invoke_cli):
+    count = 3
+
+    invoke_cli(
+        ['create', '-r', release, '-n', 'jail', '-c', count]
+    )
+
+    for i in range(1, count + 1):
+        assert jail(f'jail_{i}').exists is True, f'jail_{i} does not exist'
+
+
+@require_root
+@require_zpool
+def test_05_create_thickconfig_jail(release, jail, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'thickconfig', '-C']
+    )
+
+    jail = jail('thickconfig')
+
+    assert jail.exists is True
+    assert jail.is_thickconfig is True
+
+
+@require_root
+@require_zpool
+def test_06_create_empty_jail(jail, invoke_cli):
+    invoke_cli(
+        ['create', '-e', '-n', 'empty_jail']
+    )
+
+    jail = jail('empty_jail')
+    assert jail.exists is True
+    assert jail.is_empty is True
+
+
+@require_root
+@require_zpool
+def test_07_create_basejail(release, jail, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'basejail', '-b']
+    )
+
+    jail = jail('basejail')
+    assert jail.exists is True
+    assert jail.is_basejail is True
+
+
+@require_root
+@require_zpool
+def test_08_create_thickjail(release, jail, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'thickjail', '-T']
+    )
+
+    jail = jail('thickjail')
+    assert jail.exists is True
+    assert jail.is_thickjail is True
+
+
+@require_root
+@require_zpool
+@require_dhcp
+def test_09_test_dhcp_connectivity_in_jail(release, jail, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'dhcp_jail', 'dhcp=on'],
```

Comment:
Let's create two dhcp tests to ensure dhcp=on works and ip4_addr='vnet0|DHCP' works as well.